python/triton/runtime/jit.py

```diff
@@ -338,18 +338,42 @@ def __init__(self, module, name):
 
     def run(self, *args, grid, warmup, **kwargs):
         from ..compiler import CompiledKernel, compile, ASTSource
-        from ..compiler.backends.cuda import CUDABackend
         # deprecated arguments
         assert "device_type" not in kwargs, "device_type option is deprecated; current target will be used"
         assert "device" not in kwargs, "device option is deprecated; current device will be used"
         assert "stream" not in kwargs, "stream option is deprecated; current stream will be used"
+        def conclude_device_type(*args, **kwargs) -> str:
```

Comment:
I don't think we want that here. It is extremely expensive to run on every single kernel call (kernels may have dozens of pointer types).

Reply:
copy-paste of my slack message:

I'd actually like to only focus on GPU support for the 3.0 release if that's okay. I think supporting heterogeneous computing in Triton is a can of worms, and we may also work on our own CPU backend so we wouldn't want that to be intel-specific. If we want to support both CPUs and GPUs, then we need better torch utilities, like a set_current_device / get_current_device that also knows about CPUs. Resolving the target device type on each jit call is not reasonable, and it makes Triton super dependent on torch (what if a user passes a raw pointer that isn't a torch tensor?)